xptifw/src/xpti_trace_framework.cpp

```diff
@@ -2256,7 +2256,14 @@ class Framework {
   }
 
   uint8_t registerStream(const char *StreamName) {
-    return (uint8_t)MStreamStringTable.add(StreamName);
+    xpti::string_id_t StreamID = MStreamStringTable.add(StreamName);
+
+    // string_id_t is uint32_t while return type is uint8_t, so we need to
+    // check if the ID is valid and fits into uint8_t.
+    if (StreamID == xpti::invalid_id<xpti::string_id_t>)
```

Comment:
FYI: ` xpti::invalid_id<T>` returns the max value of the type, `T`

Reply:
`and fits into uint8_t.` that isn't checked anywhere, is it?

Reply:
I though it's implicitly assumed when the original code does `(uint8_t)MStreamStringTable.add(StreamName)`. Anyway, I've added an assert to check this assumption.